app/org/sagebionetworks/bridge/upload/DecryptHandler.java

```diff
@@ -25,6 +25,7 @@ public void setUploadArchiveService(UploadArchiveService uploadArchiveService) {
     /** {@inheritDoc} */
     @Override
     public void handle(@Nonnull UploadValidationContext context) {
-        uploadArchiveService.decryptAndUnzip(context.getStudy(), context.getData());
+        byte[] decryptedData = uploadArchiveService.decrypt(context.getStudy().getIdentifier(), context.getData());
+        context.setData(decryptedData);
```

Comment:
This would wipe out the original data.  Alternatively can the decrypted data be on its own field?  When we call context.getDecryptedData(), we will know we are reading decrypted data.